src/iotHubResourceExplorer.ts

```diff
@@ -130,27 +130,26 @@ export class IoTHubResourceExplorer extends BaseExplorer {
         });
     }
 
-    public async selectIoTHub(outputChannel: vscode.OutputChannel = this._outputChannel): Promise<IotHubDescription> {
+    public async selectIoTHub(outputChannel: vscode.OutputChannel = this._outputChannel, subscriptionId?: string): Promise<IotHubDescription> {
         TelemetryClient.sendEvent("General.Select.IoTHub.Start");
         if (!(await this.waitForLogin())) {
             return;
         }
         TelemetryClient.sendEvent("General.Select.Subscription.Start");
-        const subscriptionItems = this.loadSubscriptionItems(this.accountApi);
-        const subscriptionItem = await vscode.window.showQuickPick(subscriptionItems, { placeHolder: "Select Subscription", ignoreFocusOut: true });
-        if (subscriptionItem) {
-            TelemetryClient.sendEvent("General.Select.Subscription.Done");
+        const subscriptionItem = await this.getOrSelectSubscriptionItem(outputChannel, subscriptionId);
+        if (!subscriptionItem) {
+            return;

```

Comment:
should return or throw error?

Reply:
`return` indicates user cancelled or subscription not found.